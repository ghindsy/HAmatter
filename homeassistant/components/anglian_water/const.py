"""Anglian Water consts."""

from logging import Logger, getLogger

LOGGER: Logger = getLogger(__package__)

NAME = "Anglian Water"
DOMAIN = "anglian_water"

CONF_DEVICE_ID = "device_id"
