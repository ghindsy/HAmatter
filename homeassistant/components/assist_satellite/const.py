"""Constants for assist satellite."""

DOMAIN = "assist_satellite"
