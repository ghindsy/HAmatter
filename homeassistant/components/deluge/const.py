"""Constants for the Deluge integration."""

import enum
import logging
from typing import Final

CONF_WEB_PORT = "web_port"
CURRENT_STATUS = "current_status"
DEFAULT_NAME = "Deluge"
DEFAULT_RPC_PORT = 58846
DEFAULT_WEB_PORT = 8112
DOMAIN: Final = "deluge"
DOWNLOAD_SPEED = "download_speed"

LOGGER = logging.getLogger(__package__)
PROTOCOL_TRAFFIC_DOWNLOAD_SPEED = "protocol_traffic_download_speed"
PROTOCOL_TRAFFIC_UPLOAD_SPEED = "protocol_traffic_upload_speed"
UPLOAD_SPEED = "upload_speed"


class DelugeGetSessionStatusKeys(enum.Enum):
    """Enum representing the keys that get passed into the Deluge RPC `core.get_session_status` xml rpc method.

    you can call `core.get_session_status` with no keys (so an empty list in deluge-client.DelugeRPCClient.call)
    to get the full list of possible keys, but it seems to basically be a all of the session statistics
    listed on this page: https://www.rasterbar.com/products/libtorrent/manual-ref.html#session-statistics
    and a few others

    there is also a list of deprecated keys that deluge will translate for you and issue a warning in the log:
    https://github.com/deluge-torrent/deluge/blob/7f3f7f69ee78610e95bea07d99f699e9310c4e08/deluge/core/core.py#L58

    """

    DHT_DOWNLOAD_RATE = "dht_download_rate"
    DHT_UPLOAD_RATE = "dht_upload_rate"
    DOWNLOAD_RATE = "download_rate"
    UPLOAD_RATE = "upload_rate"
