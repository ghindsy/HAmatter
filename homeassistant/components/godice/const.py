"""Constants for the godice integration."""

from datetime import timedelta

DOMAIN = "godice"
SCAN_INTERVAL = timedelta(seconds=60)

MANUFACTURER = "Particula"
MODEL = "GoDice"

CONF_SHELL = "shell"

DICE_MIN_NUMBER = 1
DICE_MAX_NUMBER = 20
