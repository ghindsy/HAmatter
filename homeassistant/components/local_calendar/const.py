"""Constants for the Local Calendar integration."""

DOMAIN = "local_calendar"

CONF_CALENDAR_NAME = "calendar_name"
CONF_ICS_FILE = "ics_file"
CONF_IMPORT_ICS = "import_ics"
CONF_STORAGE_KEY = "storage_key"
STORAGE_PATH = ".storage/local_calendar.{key}.ics"
