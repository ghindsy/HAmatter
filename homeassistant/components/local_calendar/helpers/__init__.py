"""Helpers for local_calendar."""
