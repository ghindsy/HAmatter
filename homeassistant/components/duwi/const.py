"""Constants for the Duwi integration.

This module contains constants that are used throughout the Duwi Smart Hub integration,
including configuration keys, default values, and URLs for API endpoints.
"""

from homeassistant.const import Platform

# Unique domain identifier for the Duwi Smart Hub integration
DOMAIN = "duwi"

# Manufacturer of the product for identification purposes
MANUFACTURER = "Duwi"

# Version of the App for the Duwi Smart Hub integration, used for tracking and compatibility.
APP_VERSION = "0.1.1"

# Home Assistant client version
CLIENT_VERSION = "0.1.1"

# Model identification of the Home Assistant client, typically used for logging or diagnostics.
CLIENT_MODEL = "homeassistant"

# List of platforms that support config entry
SUPPORTED_PLATFORMS = [Platform.SWITCH]

SENSOR_TYPE = "7"
