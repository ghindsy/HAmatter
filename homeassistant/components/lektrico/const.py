"""Constants for the Lektrico Charging Station integration."""

from logging import Logger, getLogger

from homeassistant.const import Platform

# Integration domain
DOMAIN = "lektrico"

# Logger
LOGGER: Logger = getLogger(__package__)

# List the platforms that charger supports.
CHARGERS_PLATFORMS = [Platform.SENSOR]

# List the platforms that load balancer device supports.
LB_DEVICES_PLATFORMS = [Platform.SENSOR]
