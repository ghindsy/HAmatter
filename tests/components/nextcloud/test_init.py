"""Tests for the Nextcloud init."""

from syrupy.assertion import SnapshotAssertion

from homeassistant.components.nextcloud.const import DOMAIN
from homeassistant.config_entries import ConfigEntryState
from homeassistant.const import CONF_URL
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import init_integration
from .const import MOCKED_ENTRY_ID, NC_DATA, VALID_CONFIG

from tests.common import mock_registry


async def test_async_setup_entry(
    hass: HomeAssistant, snapshot: SnapshotAssertion
) -> None:
    """Test a successful setup entry."""
    entry = await init_integration(hass, VALID_CONFIG, NC_DATA)

    assert entry.state == ConfigEntryState.LOADED

    states = hass.states.async_all()
    assert len(states) == 44

    e_reg = er.async_get(hass)

    for state in states:
        assert state.state == snapshot(name=f"{state.entity_id}_state")
        assert state.attributes == snapshot(name=f"{state.entity_id}_attributes")
        assert e_reg.async_get(state.entity_id).unique_id == snapshot(
            name=f"{state.entity_id}_unique_id"
        )


async def test_unique_id_migration(hass: HomeAssistant) -> None:
    """Test migration of unique ids to stable ones."""

    entity_id = "sensor.my_nc_url_local_system_version"

    mock_registry(
        hass,
        {
            entity_id: er.RegistryEntry(
                entity_id=entity_id,
                unique_id=f"{VALID_CONFIG[CONF_URL]}#nextcloud_system_version",
                platform=DOMAIN,
                config_entry_id=MOCKED_ENTRY_ID,
            ),
        },
    )

    # test old unique id
    reg_entry = er.async_get(hass).async_get(entity_id)
    assert reg_entry.unique_id == f"{VALID_CONFIG[CONF_URL]}#nextcloud_system_version"

    await init_integration(hass, VALID_CONFIG, NC_DATA)

    # test migrated unique id
    reg_entry = er.async_get(hass).async_get(entity_id)
    assert reg_entry.unique_id == f"{MOCKED_ENTRY_ID}#system_version"
