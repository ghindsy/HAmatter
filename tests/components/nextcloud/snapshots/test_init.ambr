# serializer version: 1
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_avatars_enabled_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Avatars enabled',
  })
# ---
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_avatars_enabled_state]
  'on'
# ---
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_avatars_enabled_unique_id]
  '1234567890abcdef#system_enable_avatars'
# ---
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_debug_enabled_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Debug enabled',
  })
# ---
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_debug_enabled_state]
  'off'
# ---
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_debug_enabled_unique_id]
  '1234567890abcdef#system_debug'
# ---
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_filelocking_enabled_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Filelocking enabled',
  })
# ---
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_filelocking_enabled_state]
  'on'
# ---
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_filelocking_enabled_unique_id]
  '1234567890abcdef#system_filelocking.enabled'
# ---
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_previews_enabled_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Previews enabled',
  })
# ---
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_previews_enabled_state]
  'on'
# ---
# name: test_async_setup_entry[binary_sensor.my_nc_url_local_previews_enabled_unique_id]
  '1234567890abcdef#system_enable_previews'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_active_users_last_5_minutes_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of active users last 5 minutes',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_active_users_last_5_minutes_state]
  '0'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_active_users_last_5_minutes_unique_id]
  '1234567890abcdef#activeUsers_last5minutes'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_active_users_last_day_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of active users last day',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_active_users_last_day_state]
  '0'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_active_users_last_day_unique_id]
  '1234567890abcdef#activeUsers_last24hours'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_active_users_last_hour_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of active users last hour',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_active_users_last_hour_state]
  '0'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_active_users_last_hour_unique_id]
  '1234567890abcdef#activeUsers_last1hour'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_files_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of files',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_files_state]
  '6783'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_files_unique_id]
  '1234567890abcdef#storage_num_files'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_group_shares_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of group shares',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_group_shares_state]
  '0'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_group_shares_unique_id]
  '1234567890abcdef#shares_num_shares_groups'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_link_shares_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of link shares',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_link_shares_state]
  '2'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_link_shares_unique_id]
  '1234567890abcdef#shares_num_shares_link'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_local_storages_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of local storages',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_local_storages_state]
  '1'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_local_storages_unique_id]
  '1234567890abcdef#storage_num_storages_local'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_mail_shares_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of mail shares',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_mail_shares_state]
  '0'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_mail_shares_unique_id]
  '1234567890abcdef#shares_num_shares_mail'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_other_storages_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of other storages',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_other_storages_state]
  '1'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_other_storages_unique_id]
  '1234567890abcdef#storage_num_storages_other'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_passwordless_link_shares_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of passwordless link shares',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_passwordless_link_shares_state]
  '2'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_passwordless_link_shares_unique_id]
  '1234567890abcdef#shares_num_shares_link_no_password'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_room_shares_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of room shares',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_room_shares_state]
  '0'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_room_shares_unique_id]
  '1234567890abcdef#shares_num_shares_room'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_shares_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of shares',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_shares_received_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of shares received',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_shares_received_state]
  '1'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_shares_received_unique_id]
  '1234567890abcdef#shares_num_fed_shares_received'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_shares_sent_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of shares sent',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_shares_sent_state]
  '0'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_shares_sent_unique_id]
  '1234567890abcdef#shares_num_fed_shares_sent'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_shares_state]
  '2'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_shares_unique_id]
  '1234567890abcdef#shares_num_shares'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_storages_at_home_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of storages at home',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_storages_at_home_state]
  '2'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_storages_at_home_unique_id]
  '1234567890abcdef#storage_num_storages_home'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_storages_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of storages',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_storages_state]
  '4'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_storages_unique_id]
  '1234567890abcdef#storage_num_storages'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_user_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of user',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_user_shares_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Amount of user shares',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_user_shares_state]
  '0'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_user_shares_unique_id]
  '1234567890abcdef#shares_num_shares_user'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_user_state]
  '2'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_amount_of_user_unique_id]
  '1234567890abcdef#storage_num_users'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_apps_installed_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Apps installed',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_apps_installed_state]
  '41'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_apps_installed_unique_id]
  '1234567890abcdef#system_apps_num_installed'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cache_memory_size_attributes]
  ReadOnlyDict({
    'device_class': 'data_size',
    'friendly_name': 'my.nc_url.local Cache memory size',
    'unit_of_measurement': <UnitOfInformation.MEGABYTES: 'MB'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cache_memory_size_state]
  '0.175296'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cache_memory_size_unique_id]
  '1234567890abcdef#cache_mem_size'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cpu_load_last_15_minutes_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local CPU Load last 15 minutes',
    'unit_of_measurement': 'load',
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cpu_load_last_15_minutes_state]
  '1.45263671875'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cpu_load_last_15_minutes_unique_id]
  '1234567890abcdef#system_cpuload_15'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cpu_load_last_1_minute_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local CPU Load last 1 minute',
    'unit_of_measurement': 'load',
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cpu_load_last_1_minute_state]
  '2.06640625'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cpu_load_last_1_minute_unique_id]
  '1234567890abcdef#system_cpuload_1'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cpu_load_last_5_minutes_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local CPU Load last 5 minutes',
    'unit_of_measurement': 'load',
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cpu_load_last_5_minutes_state]
  '1.58447265625'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_cpu_load_last_5_minutes_unique_id]
  '1234567890abcdef#system_cpuload_5'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_database_size_attributes]
  ReadOnlyDict({
    'device_class': 'data_size',
    'friendly_name': 'my.nc_url.local Database size',
    'unit_of_measurement': <UnitOfInformation.MEGABYTES: 'MB'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_database_size_state]
  '4.784128'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_database_size_unique_id]
  '1234567890abcdef#database_size'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_database_type_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Database type',
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_database_type_state]
  'sqlite3'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_database_type_unique_id]
  '1234567890abcdef#database_type'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_database_version_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Database version',
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_database_version_state]
  '3.40.1'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_database_version_unique_id]
  '1234567890abcdef#database_version'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_free_memory_attributes]
  ReadOnlyDict({
    'device_class': 'data_size',
    'friendly_name': 'my.nc_url.local Free memory',
    'unit_of_measurement': <UnitOfInformation.GIGABYTES: 'GB'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_free_memory_state]
  '6.75328'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_free_memory_unique_id]
  '1234567890abcdef#system_mem_free'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_free_space_attributes]
  ReadOnlyDict({
    'device_class': 'data_size',
    'friendly_name': 'my.nc_url.local Free space',
    'unit_of_measurement': <UnitOfInformation.GIGABYTES: 'GB'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_free_space_state]
  '32.769138688'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_free_space_unique_id]
  '1234567890abcdef#system_freespace'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_free_swap_memory_attributes]
  ReadOnlyDict({
    'device_class': 'data_size',
    'friendly_name': 'my.nc_url.local Free swap memory',
    'unit_of_measurement': <UnitOfInformation.GIGABYTES: 'GB'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_free_swap_memory_state]
  '10.484736'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_free_swap_memory_unique_id]
  '1234567890abcdef#system_swap_free'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_max_execution_time_attributes]
  ReadOnlyDict({
    'device_class': 'duration',
    'friendly_name': 'my.nc_url.local PHP max execution time',
    'unit_of_measurement': <UnitOfTime.SECONDS: 's'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_max_execution_time_state]
  '3600'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_max_execution_time_unique_id]
  '1234567890abcdef#server_php_max_execution_time'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_memory_limit_attributes]
  ReadOnlyDict({
    'device_class': 'data_size',
    'friendly_name': 'my.nc_url.local PHP memory limit',
    'unit_of_measurement': <UnitOfInformation.MEGABYTES: 'MB'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_memory_limit_state]
  '536.870912'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_memory_limit_unique_id]
  '1234567890abcdef#server_php_memory_limit'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_upload_maximum_filesize_attributes]
  ReadOnlyDict({
    'device_class': 'data_size',
    'friendly_name': 'my.nc_url.local PHP upload maximum filesize',
    'unit_of_measurement': <UnitOfInformation.MEGABYTES: 'MB'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_upload_maximum_filesize_state]
  '536.870912'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_upload_maximum_filesize_unique_id]
  '1234567890abcdef#server_php_upload_max_filesize'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_version_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local PHP version',
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_version_state]
  '8.2.18'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_php_version_unique_id]
  '1234567890abcdef#server_php_version'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_system_theme_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local System theme',
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_system_theme_state]
  ''
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_system_theme_unique_id]
  '1234567890abcdef#system_theme'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_system_version_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local System version',
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_system_version_state]
  '28.0.4.1'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_system_version_unique_id]
  '1234567890abcdef#system_version'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_total_memory_attributes]
  ReadOnlyDict({
    'device_class': 'data_size',
    'friendly_name': 'my.nc_url.local Total memory',
    'unit_of_measurement': <UnitOfInformation.GIGABYTES: 'GB'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_total_memory_state]
  '30.728192'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_total_memory_unique_id]
  '1234567890abcdef#system_mem_total'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_total_swap_memory_attributes]
  ReadOnlyDict({
    'device_class': 'data_size',
    'friendly_name': 'my.nc_url.local Total swap memory',
    'unit_of_measurement': <UnitOfInformation.GIGABYTES: 'GB'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_total_swap_memory_state]
  '10.484736'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_total_swap_memory_unique_id]
  '1234567890abcdef#system_swap_total'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_updates_available_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Updates available',
    'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_updates_available_state]
  '0'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_updates_available_unique_id]
  '1234567890abcdef#system_apps_num_updates_available'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_webserver_attributes]
  ReadOnlyDict({
    'friendly_name': 'my.nc_url.local Webserver',
  })
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_webserver_state]
  'Apache/2.4.57 (Debian)'
# ---
# name: test_async_setup_entry[sensor.my_nc_url_local_webserver_unique_id]
  '1234567890abcdef#server_webserver'
# ---
# name: test_async_setup_entry[update.my_nc_url_local_none_attributes]
  ReadOnlyDict({
    'auto_update': False,
    'entity_picture': 'https://brands.home-assistant.io/_/nextcloud/icon.png',
    'friendly_name': 'my.nc_url.local None',
    'in_progress': False,
    'installed_version': '28.0.4.1',
    'latest_version': '28.0.4.1',
    'release_summary': None,
    'release_url': 'https://nextcloud.com/changelog/#28-0-4',
    'skipped_version': None,
    'supported_features': <UpdateEntityFeature: 0>,
    'title': None,
  })
# ---
# name: test_async_setup_entry[update.my_nc_url_local_none_state]
  'off'
# ---
# name: test_async_setup_entry[update.my_nc_url_local_none_unique_id]
  '1234567890abcdef#update'
# ---
